lib/camunda-platform/features/modeling/behavior/DeleteErrorEventDefinitionBehavior.js

```diff
@@ -0,0 +1,90 @@
+import inherits from 'inherits';
+
+import {
+  getBusinessObject,
+  is
+} from 'bpmn-js/lib/util/ModelUtil';
+
+import CommandInterceptor from 'diagram-js/lib/command/CommandInterceptor';
+
+const HIGH_PRIORITY = 5000;
+
+
+/**
+ * Camunda BPMN specific `camunda:errorEventDefinition` behavior.
+ *
+ * When `camunda:type` is set to something different than `external`
+ * on an element, then `camunda:errorEventDefinition` extension elements
+ * shall be removed.
+ */
+export default function DeleteErrorEventDefinitionBehavior(eventBus) {
+
+  CommandInterceptor.call(this, eventBus);
+
+  this.executed([ 'properties-panel.update-businessobject', 'element.updateProperties' ],
+    HIGH_PRIORITY, function(context) {
+      const {
+        element,
+        oldProperties,
+        properties
+      } = context;
+
+      const businessObject = getBusinessObject(element),
+            extensionElements = businessObject.extensionElements;
+
+      // (1) Check whether behavior is suitable
+      if (
+        is(element, 'camunda:ExternalCapable') &&
+        extensionElements &&
+        externalTypeChanged(oldProperties, properties)
+      ) {
+
+        // (2) Delete camunda:ErrorEventDefinition elements and save them for revert
+        context.deleteErrorEventDefinitionsOld = extensionElements.values;
```

Comment:
Just a nit-pick: `deleteErrorEventDefinitionsOld` might not be a very descriptive name. Is this a pattern we use in other places?
I would suggest something like `deletedErrorEventDefinitions`.

Reply:
Yeah, the name is a bit confusing. Let's simply use `deletedErrorEventDefinitions`, `oldExtensionElements` might be not specific enough, since this should be used by this behavior only (and not accidentally misused by others).